Sources/NIO/ByteBuffer-int.swift

```diff
@@ -105,21 +105,27 @@ extension FixedWidthInteger {
 }
 
 extension UInt32 {
-    /// Returns the next power of two unless that would overflow in which case UInt32.max is returned.
+    /// Returns the next power of two unless that would overflow, in which case UInt32.max (on 64-bit systems) or Int32.max (on 32-bit systems) is returned. The returned value is always safe to be cast to Int and passed to malloc on all platforms.
     public func nextPowerOf2ClampedToMax() -> UInt32 {
         guard self > 0 else {
             return 1
         }
 
         var n = self
 
+        #if arch(arm) // 32-bit, Raspi/AppleWatch/etc
+            let max = UInt32(Int32.max)
```

Comment:
this is maybe nit-picking but we'd like it to be `Int.max` (yes, same as `Int32.max` on 32-bit platforms) because really we'd be okay with it being larger but Swift uses the `Int` type so maybe it's clearer to users? I'm okay with leaving that too though. @Lukasa

Reply:
Fine with changing this to `Int.max`

Reply:
I'm not entirely sure about this. You want `UInt32(Int.max)`? That would fail on 64-bit platforms? This really only is for the `UInt32(Int32.max)` case, and it makes sense to be explicit here, no?

Reply:
yes `UInt32(Int.max)` would be good. I want it to fail if it's compiled on 64-bits. This thing should only be there for 32-bit platforms